### PR TITLE
fix: CVE-2021-21366

### PR DESCRIPTION
**Pull request by artemgurzhii**
https://www.npmjs.com/advisories/1650

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
     "styled_string": "0.0.1",
     "tap-parser": "^7.0.0",
     "tmp": "0.0.33",
-    "xmldom": "^0.1.19"
+    "xmldom": "0.5.0"
```
Comment (patocallaghan) [inline package.json:52]: Should this be `^0.5.0` instead considering the existing dep has `^`?

**Comment by johanneswuerbach** [comment]
The test failures look related, could you take a look?

**Comment by artemgurzhii** [comment]
@johanneswuerbach I have fixed main failing tests, but few browser tests keep failing, also I have noticed that it happens on the master branch and others, do I need to fix it or it's ok?

File: tests/ci/reporter_tests.js [hunk 2]
```diff
@@ -768,7 +768,7 @@ describe('test reporters', function() {
       reporter.finish();
       var output = stream.read().toString();
       assert.match(output, /<testsuite name="Testem Tests" tests="1" skipped="0" failures="0" timestamp="(.+)" time="(\d+(\.\d+)?)">/);
-      assert.match(output, /<testcase classname="phantomjs" name="it does &lt;cool> &quot;cool&quot; 'cool' stuff"/);
+      assert.match(output, /<testcase classname="phantomjs" name="it does <cool> &quot;cool&quot; 'cool' stuff" time="0"\/>/);
```
Comment (johanneswuerbach) [inline tests/ci/reporter_tests.js:771]: Shouldn't this actually be escaped? If I understand the test correctly having a `<` here would mean invalid XML.

Comment (artemgurzhii) [reply tests/ci/reporter_tests.js:771]: no idea actually, I have just fixed tests 🙂

Comment (johanneswuerbach) [reply tests/ci/reporter_tests.js:771]: It looks like this is actually a regression in 0.5.0 and we need to wait for 0.6.0 https://github.com/xmldom/xmldom/pull/199

Comment (johanneswuerbach) [reply tests/ci/reporter_tests.js:771]: Those changes should be reverted now :-)

Comment (artemgurzhii) [reply tests/ci/reporter_tests.js]: done, do I need to merge commits into one?

Comment (johanneswuerbach) [reply tests/ci/reporter_tests.js]: That would be appreciated, otherwise 👍

Comment (artemgurzhii) [reply tests/ci/reporter_tests.js]: done!

**Comment by johanneswuerbach** [comment]
Thank you 🙇 , published as `3.4.1`.

**Pull request merged**